cache_test.go

```diff
@@ -1302,3 +1308,18 @@ func addToCache(c *Cache[string, string], ttl time.Duration, keys ...string) {
 		}
 	}
 }
+
+func allocsForSingleRun(f func()) (numAllocs int) {
+	// `testing.AllocsPerRun` "warms up" the function for a single run before
+	// measuring allocations, so we need do nothing on the first run.
+	firstRun := false
+	numAllocs = int(testing.AllocsPerRun(1, func() {
+		if !firstRun {
+			firstRun = true
+			return
+		}
+
+		f()
+	}))
+	return
+}
```

Comment:
Following the package's naming, I think the `allocsPerSingleRun` would be a more appropriate function name. What is more, I don't see a point in using a named return here, the function scope is quite small, so something like this should be sufficient. Thanks, after this I will merge the PR. 🚀

```suggestion
func allocsPerSingleRun(f func()) int {
	// `testing.AllocsPerRun` "warms up" the function for a single run before
	// measuring allocations, so we need to do nothing on the first run.
	var firstRun bool
	
	return int(testing.AllocsPerRun(1, func() {
		if !firstRun {
			firstRun = true
			return
		}

		f()
	}))
}
```

Reply:
Hey, I’m on PTO for the next month or so, can you take over this PR please? @swithek @davseby 
If not, I’ll be to it when I’m back home :)